qiskit/circuit/equivalence.py

```diff
@@ -37,11 +35,53 @@ def __init__(self, *, base=None):
 
         Args:
             base (Optional[EquivalenceLibrary]):  Base equivalence library to
-                will be referenced if an entry is not found in this library.
+                be referenced if an entry is not found in this library.
         """
         self._base = base
 
-        self._map = {}
+        if base is None:
+            self._graph = rx.PyDiGraph()
+            self._key_to_node_index = dict()
+            self._num_gates_for_rule = dict()
+            self._rule_count = 0
+        else:
+            self._graph = copy.deepcopy(base._graph)
+            self._key_to_node_index = copy.deepcopy(base._key_to_node_index)
+            self._num_gates_for_rule = copy.copy(base._num_gates_for_rule)
+            self._rule_count = base._rule_count
+
+    @property
+    def graph(self) -> rx.PyDiGraph:
+        """Return graph representing the equivalence library data.
+
+        Returns:
+            PyDiGraph: A graph object with equivalence data in each node.
+        """
+        return self._graph
+
+    @property
+    def key_to_node_index(self) -> dict:
+        """Return map of Keys to node indices
+
+        Returns:
+            dict: A map of Keys to node indices
+        """
+        return self._key_to_node_index
+
+    @property
+    def num_gates_for_rule(self) -> dict:
+        """Return map of number of gates in each rule.
+
+        Returns:
+            dict: A map of the number of gates in each rule.
+        """
+        return self._num_gates_for_rule
```

Comment:
Not sure how to word these. I don't remember seeing any other doc strings with such warnings. I assume something like "Danger! Do not touch!" is not exactly what you're thinking.

Reply:
I think something like:

```
    """Return graph representing the equivalence library data.

    This property should be treated as read-only as it provides
    a reference to the internal state of the :class:`~.EquivalenceLibrary` object.
    If the graph returned by this property is mutated it could corrupt the
    the contents of the object. If you need to modify the output ``PyDiGraph``
    be sure to make a copy prior to any modification.
    
    Returns:
        PyDiGraph: A graph object with equivalence data in each node.
     """
```

should be sufficient. Basically just to provide some context in the docs about how the property should be used.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/8790/commits/fd5d9975b8aec810a8047b6480f99fa342a94d72.